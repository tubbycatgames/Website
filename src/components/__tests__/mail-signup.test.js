import React from "react"
import renderer from "react-test-renderer"

import MailSignup, { MailForm } from "../mail-signup"

describe("Mail Signup", () => {
  it("renders a mailing list signup form", () => {
    const tree = renderer.create(<MailSignup />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders a mailing form with an error", () => {
    const tree = renderer.create(<MailForm error={"Ooops"} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders a mailing form with a result", () => {
    const tree = renderer.create(<MailForm result={"Worked!"} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
