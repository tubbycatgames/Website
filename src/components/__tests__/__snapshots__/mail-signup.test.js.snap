// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mail Signup renders a mailing form with a result 1`] = `
<div>
  <h2
    className="header"
  >
    Join our mailing list for game updates!
  </h2>
  <p
    className="result"
  >
    Worked!
  </p>
  <form
    className="form"
  >
    <input
      aria-label="Mailing List Signup Email"
      className="input"
      name="email"
      placeholder="Enter your email here"
      required={true}
      type="email"
    />
    <button
      className="button"
      type="submit"
    >
      Sign up!
    </button>
  </form>
</div>
`;

exports[`Mail Signup renders a mailing form with an error 1`] = `
<div>
  <h2
    className="header"
  >
    Join our mailing list for game updates!
  </h2>
  <p
    className="error"
  >
    Ooops
  </p>
  <form
    className="form"
  >
    <input
      aria-label="Mailing List Signup Email"
      className="input"
      name="email"
      placeholder="Enter your email here"
      required={true}
      type="email"
    />
    <button
      className="button"
      type="submit"
    >
      Sign up!
    </button>
  </form>
</div>
`;

exports[`Mail Signup renders a mailing list signup form 1`] = `
<div>
  <h2
    className="header"
  >
    Join our mailing list for game updates!
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      aria-label="Mailing List Signup Email"
      className="input"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email here"
      required={true}
      type="email"
      value=""
    />
    <button
      className="button"
      type="submit"
    >
      Sign up!
    </button>
  </form>
</div>
`;
